common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1439,7 +1439,15 @@ def load_block_type(self, block_type):
         """
         Returns a subclass of :class:`.XBlock` that corresponds to the specified `block_type`.
         """
-        if block_type in self.disabled_xblock_types:
+        if hasattr(self, "_services") and 'request_cache' in self._services:
```

Comment:
Sadly, in this particular instance, I'm not sure there is a better way. You don't have an XBlock to use the `.service` method with.

I guess the other option would be to embed the request-caching behavior into the fetch_disabled_xblock_types method, since at the time that you're defining that, you're in `xmodule/django.py`, which is allowed to use django-specific things, and already has access to the request cache.